### PR TITLE
Add default enclave_type flag for oe_create_enclave

### PR DESCRIPTION
**Pull request by paulcallen**
Instead of requiring a specific type to be specified, like
OE_ENCLAVE_TYPE_SGX a developer can now specify OE_ENCLAVE_TYPE_AUTO
which will pick the enclave type based on the platform that is being
built. Therefore x64 will assume SGX, where as ARM will assume
trustzone (when the code is merged!).
This is part of the fix for #844, the less contentious part!

### CODE REVIEW

File: tests/create-errors/host/host.c [hunk 1]
```diff
@@ -19,9 +19,9 @@ static void _test_invalid_param(const char* path, uint32_t flags)
 
     OE_TEST(result == OE_INVALID_PARAMETER);
 
-    /* Invalid enclave type. */
+    /* Invalid enclave type. Note that 0 is now allowed! */
     result = oe_create_create_errors_enclave(
-        path, OE_ENCLAVE_TYPE_UNDEFINED, flags, NULL, 0, &enclave);
+        path, __OE_ENCLAVE_TYPE_MAX, flags, NULL, 0, &enclave);
```
[the remark this answers is not in the document]
Comment (paulcallen) [reply tests/create-errors/host/host.c]: done

File: include/openenclave/bits/types.h [hunk 2]
```diff
@@ -80,13 +80,17 @@ typedef long long ptrdiff_t;
...
+ * use SGX.
+ * **OE_ENCLAVE_TYPE_SGX** will force the platform to use SGX, but any platform
+ * other than x64 will not support this and will generate errors.
  */
 typedef enum _oe_enclave_type
 {
-    OE_ENCLAVE_TYPE_UNDEFINED,
+    OE_ENCLAVE_TYPE_AUTO,
     OE_ENCLAVE_TYPE_SGX,
+    OE_ENCLAVE_TYPE_UNDEFINED,
```
Comment (dthaler) [inline include/openenclave/bits/types.h]: When would an app use UNDEFINED instead of AUTO?   Trying to figure out why it would have a different value.

Comment (paulcallen) [reply include/openenclave/bits/types.h]: It should never be used. It is an invalid value. I am open to renaming this if there is a better name.

Comment (dthaler) [reply include/openenclave/bits/types.h]: If it should never be used, and you're making a breaking change anyway, then it should simply be removed from the enum, not added as another new value that should never be used.

Comment (dthaler) [reply include/openenclave/bits/types.h]: The other alternative is to leave it as a deprecated alias for AUTO, along the lines of
#ifdef OE_V1_APIS
# define OE_ENCLAVE_TYPE_UNDEFINED OE_ENCLAVE_TYPE_AUTO
#endif

**Pull request merged**